work/worker.go

```diff
@@ -907,7 +905,7 @@ func (env *Task) commitBundleTransaction(bundle *builder.Bundle, bc BlockChain,
 	}
 
 	restoreEnv := func() {
-		*env.state = *lastSnapshot
+		env.state = lastSnapshot
```

Comment:
Just a question: How did it lead to different gas used?

Reply:
I'm not sure exactly, but I guess some member variables of `env.state` are not completely restored by the assignment.

Reply:
It might be related to cold storage access since the gas difference is close to 2100.

Maybe when tx has been reverted, the `env.state` will remember the context of reverted tx, so the next tx considers it warm storage, while validators consider it cold storage. I think it's the most important part, so it's worth investigating to find the root cause. Please share your thoughts. Thanks!

Reply:
I confirmed that bad block occurs when shallow-copying state objects as follows:

```
func copyStateDB(dst, src *StateDB) {
	...
	dst.stateObjects = src.stateObjects
	dst.stateObjectsDirty = src.stateObjectsDirty
	dst.stateObjectsDirtyStorage = src.stateObjectsDirtyStorage
}
```

Reply:
fyi, `maps.Clone` is a shallow copy.

Reply:
Does it also fix the copy-of-copy issue?

Reply:
What is the issue? Is it related to this test? 
https://github.com/kaiachain/kaia/blob/beb73629c0bf41af1f07100f416448daf5d7dc5c/blockchain/state/statedb_test.go#L139

Reply:
Isn't `Two deep-copy of statedb upon bundle fail (c.f. one deep-copy upon bundle success).` the issue? I saw this in the scrum today.

Reply:
Oh, I meant upon bundle failure, two deep-copies (taking a snapshot, and restoring the old object) take place. Sorry for the confusion

Reply:
Oh, I misunderstood it. Thanks!